OMCompiler/Compiler/BackEnd/CommonSubExpression.mo

```diff
@@ -1013,11 +1029,39 @@ algorithm
         end if;
      then isRedundant;
 
+    // (a,b) = (c,d)
     case BackendDAE.COMPLEX_EQUATION(left=DAE.TUPLE(lhs), right=DAE.TUPLE(rhs)) guard (listLength(lhs) == listLength(rhs))
       algorithm
         (globalKnownVarHT, globalKnownVars, orderedVars, isRedundant) := isEquationRedundant_flatten2(lhs, rhs, globalKnownVarHT, globalKnownVars, orderedVars);
      then isRedundant;
 
+  // (a,b) = c
+  case BackendDAE.COMPLEX_EQUATION(_, exp1 as DAE.TUPLE(lhs), exp2, _, _)
+    algorithm
+      isRedundant := Expression.expEqual(exp1, exp2);
+      if not isRedundant then
+        isGlobalKnown := allArgsInGlobalKnownVars({exp2}, globalKnownVarHT);
+        if isGlobalKnown then
+          for expMem in lhs loop
+            // create variable with bind exp
+            varList := createVarsForExp(expMem, {});
+            for var in varList loop
+              // Add var to globalKnownVars
+              var := BackendVariable.setBindExp(var, SOME(exp2));
+              globalKnownVars := BackendVariable.addVar(var, globalKnownVars);
+              // Add cref(s) to globalKnownVarHT
+              globalKnownVarHT := addConstantCseVarsToGlobalKnownVarHT(expMem, globalKnownVarHT);
+              // Delete var from ordered vars if no cse-cref
+              if not isCSECref(var.varName) then
+                // TODO: Do I need to delete stuff here?
```

Comment:
Do they need to be CSE vars? If you have
```
(name1, name2) = func();
```

or 


```
(cse1, cse2) = func();
```

does not make a difference right? And you only get to this case if both elements are records and not any expression?

Maybe we want to generalize this to tuples with any number of elements?

Reply:
I guess I can delete [lines 1055-1058](https://github.com/OpenModelica/OpenModelica/pull/6999/files#diff-9cc4431be7f701499677dfee1e848acbe3beb830f1fe4fa16b25514b55ebbe12R1055).
In `isEquationRedundant_flatten2` there is something similar done, so I used it as a reference. But for me those lines removed a variable to much (the ones that don't have a alias).

The case for tuples is handled [above](https://github.com/OpenModelica/OpenModelica/pull/6999/files#diff-9cc4431be7f701499677dfee1e848acbe3beb830f1fe4fa16b25514b55ebbe12R1033).
They can only have the same size or the right size is a function call, right?
or what could be more general?

Reply:
Sounds good. I could think of weird things that could be modelica language viable but nonsense from a modeling standpoint:
```
((a, b), c) = (func1(), func2());
```
with `func1` returning two and `func2` returning one record. I don't know if people do stuff like this but i hope not.

Reply:
I think that is not allowed. At least I tried with a func that returns only one record:
```modelica
(recordA_1,recordA_2) = (func(in1),func(in2));
```

Reply:
Yeah, tuples can only be used on the form `(a, b, c) = func(...)`, see [specification](https://specification.modelica.org/master/equations.html#simple-equality-equations).

Reply:
I think the grammar, at least, allows you to write 

`(a, b, c) + (1, 2, 3) `

You can also have 
`(a+b, c+4) = func(...) `

😊

Reply:
Oops, non-cref tuples fixed in #7013. `(a, b, c) + (1, 2, 3)` is already forbidden by both frontends.

Reply:
That speed! 😂.

Modelica needs to fix it's grammar too. It is not your fault. You can have all kinds of combinations because` (exp,exp....) `is allowed as a primary expression.

```
primary : "(" output-expression-list ")"

output-expression-list :
   [ expression ] { "," [ expression ] } 
```

https://specification.modelica.org/master/modelica-concrete-syntax.html#expressions1

Reply:
Actually, thinking about it, they don't have to "fix" it. Of course it can always be stated as a semantic error.

Still I do not know why `"(" output-expression-list ")"` is part of primary. It is not allowed anywhere except for the form  `(a,b,c) = func(...)` as you said.